monitor/api/lineage/schema.py

```diff
@@ -16,19 +16,16 @@ class NodeDependsOnNodesSchema(BaseModel):
     @validator("depends_on_nodes", pre=True, always=True)
     def set_depends_on_nodes(cls, depends_on_nodes):
         formatted_depends_on = depends_on_nodes or []
-        return [cls._format_node_id(node_id) for node_id in formatted_depends_on]
-    
-    @validator("unique_id", pre=True, always=True)
-    def set_unique_id(cls, unique_id):
-        return cls._format_node_id(unique_id)
-    
+        formatted_depends_on = [cls._format_node_id(node_id) for node_id in formatted_depends_on]
+        return [node_id for node_id in formatted_depends_on if node_id]
+
     @classmethod
     def _format_node_id(cls, node_id: str):
-        # Currently seeds are saved as sources.
-        # Convert seed prefix into source for nodes unique ids.
+        # Currently we don't save seeds in our artifacts.
+        # We remove seeds from the lineage graph (as long as we don't support them).
         seed_patern = re.compile(r"^seed\.")
```

Comment:
Maybe just check if the if the node_id starts with seed? because that's what this regex essentially validates. If you still want to keep the regex, move the compile outside of the function to be defined at the top of the class instead of in every invocation

Reply:
Good point.
I like using regex, it is clear and super stricted :)
Will move the compile up!